From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: EWouters <6179932+EWouters@users.noreply.github.com>
Date: Tue, 5 Apr 2022 14:33:34 +0200
Subject: [PATCH] Patch Makefile

- Make `CC` configurable from `env`
- Remove `klong.image` target from all
- Add install target
---
 Makefile | 9 +++++++--
 1 file changed, 7 insertions(+), 2 deletions(-)

diff --git a/Makefile b/Makefile
index f692ff7..470adcd 100644
--- a/Makefile
+++ b/Makefile
@@ -1,5 +1,5 @@
 CFLAGS=	-g -Wall -pedantic -O3
-CC=	cc $(CFLAGS)
+CC:=	$(CC) $(CFLAGS)
 V=	20211014
 R=	20220315
 
@@ -12,7 +12,7 @@ OPTIONS= -DEDIT
 # Modules to load into image file
 MODULES=-l nstat -l nplot -l time
 
-all:	kg klong.image
+all:	kg
 
 klong.image:	kg
 	./kg -n $(MODULES) -o klong.image
@@ -54,3 +54,8 @@ csums:
 
 mksums:	clean
 	find . -type f | grep -v _csums | grep -v klong2015 | csum >_csums
+
+install:
+	mkdir -p ${DESTDIR}/usr/local/bin
+	install kg ${DESTDIR}/usr/local/bin
+	mkdir -p ${DESTDIR}/usr/local/lib/klong
+	install -m 644 lib/* ${DESTDIR}/usr/local/lib/klong
